### PR TITLE
[pigeon] fixes Java generator uses wrong style for enum values

### PR DESCRIPTION
**Pull request by cofirazak**
Pigeon Java code generator uses wrong style (camelCase from Dart) for enum values. This PR fixes it to UPPER_CASE naming convention for enum values.

Fixes [#105801](https://github.com/flutter/flutter/issues/105801)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by cofirazak** [comment]
@stuartmorgan 
IDKW, but there was no reviewers added automatically to this PR.
And i don't see how can i add someone manually.

**Comment by stuartmorgan** [comment]
There's no auto-assignment set up for this repository. We do a weekly triage and route to reviewers that way.

File: packages/pigeon/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 3.2.7
+
+* [java] fixes generator uses wrong style (camelCase) for enum values.
```
Comment (gaaclarke) [inline packages/pigeon/CHANGELOG.md:3]:
```suggestion
* [java] Changes style for enum values from camel-case to snake-case.
```

**Review by gaaclarke — CHANGES_REQUESTED** [review]
The code looks good to me.  I think we need a breaking change notification in the CHANGELOG too.  I'm not sure if we should version it as a breaking change.  I can see arguments both ways, thoughts @stuartmorgan?

File: packages/pigeon/test/java_generator_test.dart [hunk 2]
```diff
@@ -41,7 +41,7 @@ void main() {
       name: 'Foobar',
       members: <String>[
         'one',
-        'two',
+        'twoThreeFour',
```
Comment (stuartmorgan) [inline packages/pigeon/test/java_generator_test.dart:44]: We should also include a test with something that follows the Dart style guide camel case exception for two-letter acronyms. E.g., `remoteDB`.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Yes, we should version this as a breaking change; it requires someone using a non-experimental generator to change the code they've already written, which we've been treating as a breaking change. The generated interfaces are effectively the API of this package.

**Review by stuartmorgan — APPROVED** [review]
LGTM

@gaaclarke for second review

**Review by gaaclarke — APPROVED** [review]

**Pull request merged**